## Enviando E-Mails

* SMTP = *Simple Mail Transfer Protocol*.


* Define como os e-mails devem ser enviados.

In [1]:
import smtplib

## Configurar o provedor de e-mail

O nome de domínio do servidor SMTP normalmente será o nome de domínio de seu provedor de emails com SMTP. na frente.

In [2]:
smtpObj = smtplib.SMTP('smtp.gmail.com', 587)

## Testar a conexão

In [3]:
smtpObj.ehlo()

(250,
 b'smtp.gmail.com at your service, [2804:14d:1484:4256:8054:2d61:afb5:87d6]\nSIZE 35882577\n8BITMIME\nSTARTTLS\nENHANCEDSTATUSCODES\nPIPELINING\nCHUNKING\nSMTPUTF8')

* Se retornou `250`, é porque está tudo OK.

## Precisamos deixar nossa aplicação segura utilizando criptografia TLS

In [4]:
smtpObj.starttls()

(220, b'2.0.0 Ready to start TLS')

 * O `220` no valor de retorno informa que o servidor está pronto.

## Vamos realizar login

In [5]:
remetente = 'exemplo.python@gmail.com'

In [6]:
smtpObj.login(remetente, 'exemplo123')

(235, b'2.7.0 Accepted')

## Agora iremos enviar nosso primeiro e-mail com Python

In [8]:
body = 'Subject: So long.\nDear Alice, so long and thanks for all the fish. Sincerely, Bob'

destinatario = 'email' # Preencha o e-mail do destinatário aqui

smtpObj.sendmail(remetente, destinatario,  body)

{}

## É bem simples essa tarefa. Mas se eu quiser enviar e-mails para mais de 1 pessoa?

In [9]:
mailList = ['email_1', 
            'email_2'] # Preencha os e-mails dos destinatários aqui

body = 'Subject: So long.\nDear Alice, so long and thanks for all the fish. Sincerely, Bob'

for mail in mailList:
    smtpObj.sendmail(remetente, mail,  body)

## Iremos realizar uma pequena modificação. Agora, a minha lista tem o nome da pessoa e o seu e-mail

In [10]:
mailDict = {'name_1' : 'email_1', 
            'name_2' : 'email_2'} # Preencha os nomes e e-mails dos destinatários aqui

subject = 'Subject: So long.\n'

for name, mail in mailDict.items():
    body = subject + 'Dear ' +  name + ', so long and thanks for all the fish. Sincerely, Bob'
    smtpObj.sendmail(remetente, mail,  body)

## Após isso, iremos fechar a conexão

In [11]:
smtpObj.quit()

(221, b'2.0.0 closing connection w4-v6sm1451139uaj.8 - gsmtp')